src/creatures/creature.cpp

```diff
@@ -998,7 +998,9 @@ BlockType_t Creature::blockHit(std::shared_ptr<Creature> attacker, CombatType_t
 		attacker->onAttackedCreatureBlockHit(blockType);
 	}
 
-	mitigateDamage(combatType, blockType, damage);
+	if (damage > 0) {
```

Comment:
I believe you can simply remove this id and add the code inside the if (damage != 0) that comes right below